### ISSUE TITLE
X3D-Edit autoinstaller and zip for Windows fails, no workaround

### ISSUE THREAD

**Issue by brutzman** [apache/netbeans#6934]
### Apache NetBeans version

Apache NetBeans 20

### What happened

Creating autoinstaller for X3D-Edit module does not find local Java once installed.  Workaround is to edit netbeans.conf file, which is typically difficult for end user.

Details found at
- https://github.com/Web3DConsortium/X3D-Edit/issues/3
 

### Language / Project Type / NetBeans Component

NetBeans Windows Installer

### How to reproduce

Example of autoinstaller exhibiting this error are found at
- https://sourceforge.net/projects/x3d/files/X3D-Edit%20Pre-Release%20Testing/x3deditmodulesuite-windows.exe/download

Can download source for X3D-Edit module and then test autogeneration using the source archive
- https://github.com/Web3DConsortium/X3D-Edit

### Did this work correctly in an earlier version?

No. This has never worked properly for us in previous versions of NetBeans over the past several years.  Possibly we have misconfigured something.

### Operating System

OS Name	Microsoft Windows 10 Pro, Version 10.0.19045 Build 19045

### JDK

OpenJDK 21.0.1

### Apache NetBeans packaging

Apache NetBeans provided installer

### Anything else

We are happy to apply modifications and test further.

### Are you willing to submit a pull request?

No thanks, we are not building or modifying NetBeans source.

**Comment by neilcsmith-net** [comment]
What does creating an autoinstaller for X3D-Edit mean exactly?  Are you talking about distribution of a NetBeans platform application containing the module?  For the latter I would recommend building an installer with [NBPackage](https://github.com/apache/netbeans-nbpackage/) and ideally shipping with bundled JDK / Java runtime.

The Windows installer infrastructure, particularly for platform applications, is basically unmaintained.

Not sure why you mention `netbeans.conf` as opposed to `<branding>.conf`?  Anyway, the other option with either is to edit the desktop shortcut to added the JDK location.

**Comment by neilcsmith-net** [comment]
There is a known issue with the Windows launcher not finding Java (assuming it's installed in a discoverable way).  Hopefully that can be fixed in future.  However, you cannot exactly guarantee if that's set up correctly, or that it points to what you need.  Relying on a system-wide Java install is really something for the past.

Also, as said above, no-one is maintaining the package as installers or macos application.  They are almost untouched since donation to ASF.  NetBeans itself has moved away from using that installer infrastructure for anything other than Windows, and we are looking at replacing that (actually pending on some launcher fixes).

If your overall goal is a trouble free experience working out-of-the-box, I would suggest looking at shipping the JDK (or stripped down runtime depending on what you need), and including that with NBPackage as outlined above.

**Comment by brutzman** [comment]
Acknowledged with thanks, we'll think about when we can pursue that direction.

Suggest considering changes to NetBeans "_Package as_" menu actions so that developers are alerted to issues when trying them.  Having a help page on all this would be helpful too.

**Comment by neilcsmith-net** [comment]
There are two issues here.  The one about the NBI Windows installer is not being fixed.  Discussion is ongoing on removing that option on the dev@ list at the moment https://lists.apache.org/thread/wt6zpl0ll1vg959soo7gfz7pl8bdyccl  The correct solution has already been given.

The other issue is that the Windows launcher cannot find a JDK, and so the zip package fails to launch.  That is a known issue, and would be good to get fixed.  That is also discussed in the same thread.

**Comment by neilcsmith-net** [comment]
@brutzman I'm not sure why you expected anything different?! The solution has been given, most of this is a won't fix, and the UI changes (removal) are under discussion.

**Comment by brutzman** [comment]
Thanks @neilcsmith-net - I was not expecting anything different, and am grateful for the deliberate pace of progress throughout NetBeans.  It so happens that we had a new end user trouble report about this issue last week.  I add the confirmation and version numbers for others, to be clear that this was still an open issue (until resolved or removed).  Appreciate the clarity of closing it as not planned, we will update our X3D-Edit documentation accordingly.
